### DISCUSSION TITLE
Repeat scheduled task specific number of times

### DISCUSSION THREAD

**Discussion by ekainz** [TcMenu/TaskManagerIO#49] [Q&A] [answered]
I'd like to drive a servo using TaskManager but am having a hard time conceptualizing how to translate my code. My current control code is below:

```
  for(int x = 0; x < steps; x++){
    digitalWrite(pinMotorZStep,HIGH); 
    delayMicroseconds(delayTime); 
    digitalWrite(pinMotorZStep, LOW); 
    delayMicroseconds(delayTime);
  }
```
  
  How could I approach this with TaskManagerIO?

**Comment by davetcc** [comment]
Hi there the easiest way to achieve that is to schedule something once and use a counter to count down.

ie the idea would be you have a function that reschedules itself until the condition is met. A bit like setTimeout in JavaScript. We do this quite a bit in tcMenu. 

[EDIT] correct auto-correct spelling

**Reply by davetcc** [reply, marked as answer]
Here's the common pattern for that case, its used quite a bit in TcMenu and elsewhere:

```
// somewhere in your code
taskManager.execute(taskManagerFn);

// and the function looks like
void taskManagerFn() {
     // do some work

    if(stillNeedToScheduleAgain) {
        taskManager.scheduleOnce(interval, taskManagerFn);
    }
}
```

**Reply by davetcc** [reply]
It looks like you're trying to use this for stepper control, and I can tell you that nothing other than interrupt-based routines will be fast enough for that, as any rendering to the display or other tasks will take too long. Depending on the board, you could try running the work on another thread (IE a second loop), but I doubt even that would be accurate enough for stepper control. 

I think this is one of the few cases where if you want to do anything else than just control the stepper, you really need to use an interrupt-based servicing routing.
